axoupdater/src/test/helpers.rs

```diff
@@ -0,0 +1,114 @@
+use std::{
+    env::consts::EXE_SUFFIX,
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+
+static RECEIPT_TEMPLATE: &str = r#"{"binaries":[BINARIES],"install_prefix":"INSTALL_PREFIX","provider":{"source":"cargo-dist","version":"0.10.0-prerelease.1"},"source":{"app_name":"APP_NAME","name":"PACKAGE","owner":"axodotdev","release_type":"github"},"version":"VERSION"}"#;
+
+/// Generates an install receipt given the specified fields and returns it as a string.
+pub fn install_receipt(
+    app_name: &str,
+    package: &str,
+    binaries: &[&str],
+    version: &str,
+    prefix: &str,
+) -> String {
+    let binaries = binaries
+        .iter()
+        .map(|name| format!(r#""{name}""#))
+        .collect::<Vec<String>>()
+        .join(", ");
+    RECEIPT_TEMPLATE
+        .replace("BINARIES", &binaries)
+        .replace("PACKAGE", package)
+        .replace("APP_NAME", app_name)
+        .replace("INSTALL_PREFIX", &prefix.replace('\\', "\\\\"))
+        .replace("VERSION", version)
+}
+
+/// Generates an install receipt given the specified fields and writes it to disk at the appropriate location in `config_path`. The path to the new file is returned.
+pub fn write_receipt(
+    app_name: &str,
+    package: &str,
+    binaries: &[&str],
+    version: &str,
+    prefix: &PathBuf,
+    config_path: &PathBuf,
+) -> std::io::Result<PathBuf> {
+    let contents = install_receipt(
+        app_name,
+        package,
+        binaries,
+        version,
+        &prefix.to_string_lossy(),
+    );
+    let receipt_name = config_path.join(format!("{package}-receipt.json"));
+    std::fs::create_dir_all(&config_path)?;
+    std::fs::write(&receipt_name, contents)?;
+
+    Ok(receipt_name)
+}
+
+/// Performs an end-to-end run test for `app_name` within `package`. `bin` is the path to the executable being tested, usually the one from the `CARGO_BIN_EXE_<name>` environment variable.
+/// Rather than returning a value, this function performs assertions.
+/// Because it writes to real files outside a temporary directory, it's highly recommended that this only becalled within CI builds.
```

Comment:
This could use some reformatting, the args here are complex enough to want to really spell them out

Reply:
Would also love an example here.

Reply:
I'm going to turn this into a struct so we get "named args" that's clearer.